### PR TITLE
Db/synchandler doclets

### PR DESCRIPTION
**Pull request by goransle**

### CODE REVIEW

File: ts/Dashboards/Components/Sync/Emitter.ts [hunk 1]
```diff
@@ -18,41 +18,78 @@ import type ComponentTypes from '../ComponentType';
...
+     * Registry for reusable emitter.
+     * The emitter is stored by ID.
      */
     public static registry: Record<string, SyncEmitter> = {};
 
-    public static register(handler: SyncEmitter): void {
-        const { id } = handler;
-        this.registry[id] = handler;
+    /**
+     * Adds an emitter to the emitter regisitry.
```
Comment (karolkolodziej) [inline ts/Dashboards/Components/Sync/Emitter.ts:32]:
```suggestion
     * Adds an emitter to the emitter registry.
```

**Review by karolkolodziej — CHANGES_REQUESTED** [review]

**Comment by karolkolodziej** [comment]
Looks good but I'd like to know if we should show all of that.

IMO the `OptionsEntry` should be enough for now especially after we establish that there should not be any standalone components.

**Review by pawelfus — COMMENTED** [review]
+1 to what Karol said: how much of it is necessary to build a dashboard? Which other part is necessary to build custom components and create sync between those and existing ones?

**Review by sebastianbochan — APPROVED** [review]

**Comment by goransle** [comment]
> +1 to what Karol said: how much of it is necessary to build a dashboard? Which other part is necessary to build custom components and create sync between those and existing ones?

`OptionsEntry` should cover it it for most use cases.

**Review by pawelfus — APPROVED** [review]
Thanks!

**Review by bre1470 — APPROVED** [review]
Looks good. 👍

**Review by pawellysy — APPROVED** [review]

**Review by karolkolodziej — APPROVED** [review]

**Pull request merged**